File: scala-advent-of-code/src/test/scala/advent/twenty_sixteen/Day06SignalsAndNoiseTest.scala
package advent.twenty_sixteen

import org.scalatest.{FlatSpec, Matchers}

class Day06SignalsAndNoiseTest  extends FlatSpec with Matchers {
  "2016 Day 06 - Input" should "calculate answers" in {
    val dp = new Day06SignalsAndNoise("/twenty_sixteen/Day06-SignalsAndNoise-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day06SignalsAndNoise("/twenty_sixteen/Day06-SignalsAndNoise-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}

File: java-advent-of-code/src/test/java/advent/twenty_sixteen/Day06SignalsAndNoiseTest.java
package advent.twenty_sixteen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day06SignalsAndNoiseTest {
    @Test
    public void checkReal() {
        DailyProblem<String, String> dp = new Day06SignalsAndNoise("/twenty_sixteen/Day06-SignalsAndNoise-input.txt");

        assertEquals("gebzfnbt", dp.getPart1Answer());
        assertEquals("fykjtwyn", dp.getPart2Answer());
    }

    @Test
    public void checkExample1() {
        DailyProblem<String, String> dp = new Day06SignalsAndNoise("/twenty_sixteen/Day06-SignalsAndNoise-example#1.txt");

        assertEquals("easter", dp.getPart1Answer());
        assertEquals("advent", dp.getPart2Answer());
    }
}